packages/fxa-react/components/DataBlock/index.tsx

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, {
+  useCallback,
+  useState,
+  useEffect,
+  MouseEvent,
+  FocusEvent,
+} from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { copy } from 'fxa-react/lib/clipboard';
+
+export type DataBlockProps = {
+  value: string | string[];
+  copyable?: boolean;
+  id?: string;
+  onCopied?: (copiedValue: string) => void;
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
+  onBlur?: (event: FocusEvent<HTMLButtonElement>) => void;
+  onFocus?: (event: FocusEvent<HTMLButtonElement>) => void;
+  readerText?: string;
+};
+
+export const DataBlock = ({
+  value,
+  id,
+  copyable,
+  onCopied,
+  onClick,
+  onBlur,
+  onFocus,
+  readerText,
+}: DataBlockProps) => {
+  const [focussed, setFocussed] = useState<boolean>(false);
+  const valueIsArray = Array.isArray(value);
+  const [inputId, setInputId] = useState<string | undefined>(id);
+  const readerId = `${inputId}-sr`;
+
+  async function copyToClipboard() {
+    const copyValue = valueIsArray
+      ? (value as string[]).join(', ')
+      : (value as string);
+
+    await copy(copyValue);
+    onCopied && onCopied(copyValue);
+  }
+
+  useEffect(() => {
+    !inputId && setInputId(uuidv4());
+  }, [inputId, id]);
+
+  const dataBlockClick = useCallback(
+    (event: MouseEvent<HTMLButtonElement>) => {
+      copyable && copyToClipboard();
+      onClick && onClick(event);
+    },
+    [focussed]
+  );
+
+  const dataBlockFocus = useCallback(
+    (event: FocusEvent<HTMLButtonElement>) => {
+      setFocussed(true);
+      onFocus && onFocus(event);
+    },
+    [focussed]
+  );
+
+  const dataBlockBlur = useCallback(
+    (event: FocusEvent<HTMLButtonElement>) => {
+      setFocussed(false);
+      onBlur && onBlur(event);
+    },
+    [focussed]
+  );
+
+  return (
+    <>
+      <button
+        className={`flex rounded-xl px-7 font-mono text-sm text-green-900 bg-green-800 bg-opacity-10 flex-wrap ${
+          valueIsArray ? 'py-4' : 'py-5'
+        } ${
+          copyable
+            ? 'hover:bg-opacity-20 focus:bg-opacity-30 active:bg-opacity-30'
+            : 'cursor-default select-text'
+        }`}
+        aria-describedby={readerText ? readerId : undefined}
```

Comment:
I'm pretty sure this can't be `readerText`, and I'm not sure if we need this. `aria-describedby` should be the ID associated. An `aria-label` will read text if we need to do that instead, but you can have `aria-label`, `aria-describedby`, _and_ `aria-labelledby` if we need it. I think `aria-describedby` and `aria-labelledby` give enough context here because then the label will be read in addition. What use case do you think we may have?

Reply:
Edit: oops I looked at this incorrectly, it's only ever `readerId` or `undefined`, so ignore the first sentence.

Reply:
Slight confusion per our Slack chat, BUT, what if I remove all the describedby stuff and just allowed all the components to receive it as a prop, leaving it up to the dev in each specific use case if they want to add additional input descriptors?